### PR TITLE
fix authconfig hosts when targeting gateway

### PR DESCRIPTION
**Pull request by eguzki**

### CODE REVIEW

**Review by guicassolato — APPROVED** [review]

**Review by guicassolato — APPROVED** [review]
2 bug fixes + new test case. Nice job, @eguzki!

**Comment by guicassolato** [comment]
> 2 bug fixes + new test case. Nice job, @eguzki!

Oops. Too soon.

>  GitHub Actions / Integration Tests
> 
> Expect(gateway.Spec.Listeners).Must undefined (type "github.com/onsi/gomega/types".Assertion has no field or method Must)

**Review by didierofrivia — APPROVED** [review]
🎖️  🤟🏼

File: controllers/authpolicy_controller_test.go [hunk 1]
```diff
@@ -132,6 +129,52 @@ var _ = Describe("AuthPolicy controller", func() {
...
+					AuthScheme: testBasicAuthScheme(),
+				},
+			}
+
+			err = k8sClient.Create(context.Background(), policy)
+			logf.Log.V(1).Info("Creating AuthPolicy", "key", client.ObjectKeyFromObject(policy).String(), "error", err)
+			Expect(err).ToNot(HaveOccurred())
+
+			// check policy status
+			Eventually(testPolicyIsReady(policy), 60*time.Second, 5*time.Second).Should(BeTrue())
```
Comment (didierofrivia) [inline controllers/authpolicy_controller_test.go:164]: Mhmm, maybe we can't rely on the time frame provided

**Comment by eguzki** [comment]
not some auth policy controllers are failing :facepalm:

**Comment by eguzki** [comment]
Waiting https://github.com/Kuadrant/kuadrant-operator/pull/303 to be merged to rebase this.

**Comment by alexsnaps** [comment]
@eguzki #303 is merged

File: controllers/authpolicy_controller_test.go (hunk 1 above)
Comment (eguzki) [reply controllers/authpolicy_controller_test.go:174]: the check is correct. It was rightfully saying something is wrong. 

I missed adding a route. So a policy targeting a gateway without routes do not have any effect and the policy is not ready.

**Review by guicassolato — APPROVED** [review]

**Pull request merged**